correlation_id/config.go

```diff
@@ -11,11 +11,37 @@ type Config struct {
 	IdGenerator func(*http.Request) string
 }
 
-func NewConfig() *Config {
-	return &Config{
+func (c *Config) apply(options ...Option) *Config {
+	for _, option := range options {
+		option(c)
+	}
+	return c
+}
+
+// GetConfig return a new correlation configuration with all options applied
+func GetConfig(options ...Option) *Config {
```

Comment:
maybe declare this public func before the private `apply` func ?

Reply:
i really prefer group object with his method and put the new func just after